packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+
+```
+Upgrade: Bump packageA to 2.0.0
+New: This shiny new feature
+```
+
+instead of the reverse. Because of the way that monorepos work this is actually expected. Also this will be "hidden" from the changelog as `Upgrade:` and `Chore:` are not outputed in the process.
+
+### Breaking change
+
+* `packageA` breaks something → `Breaking: Drastic change in the API`.
+* `packageB` changes also needs to change completely to make sure how everything works → `Breaking: Drastic change in the API`.
+
+The release script will:
+* releases `packageA` with a `major` bump: version 2.0.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 2.0.0`. Because the commit history has `Breaking` and `Upgrade`, `packageB` will be bumped to 2.0.0 (`Breaking` > `Upgrade`).
+
+E.g.: If `hint` changes the expectations of the resources should look like, updating will be a `Breaking`.
+
+**⁉️ NOTE**: `packageA` commit could be anything, the important part is how the changes are handled in `packageB` and how they are commited. It is very tempting to do all the changes in one single commit but this could have very bad consequences (e.g.: bumping to a `major` all packages when it should be a `patch`).
+
+## Current release process
+
+The previous is implemented (more or less accurate) during the release process. To start a new release a use needs to run `npm run release` that executes `scripts/release.ts`. The gist is:
+
+1. Calculate what packages need to be published. To do this searches the latest tag for each package and checks if there have been any commits since then that affect it.
+1. Sort the dependencies by less dependants, i.e. if `packageA` depends on `packageB` it will work on `packageB` first.
+1. For each package:
+   * Bump the `version` in the `package.json` of the package usint the related commit message(s) and update `CHANGELOG.md`
+   * Clean the package, intall dependencies via `npm`, and run `npm run test-release`
+   * Publish the new package to npm
+   * Update references from other packages
+   * Publish changes into GitHub
+
+This script also takes care of removing the `private` property from `package.json`, initialize a new `CHANGELOG.md` if needed, download files that are updated elsewhere like snyk's database, and such.
+
+## Current problems
+
+While the current process works most of the times OK, there are a few things that could be improved:
+
+* Using `tag`s to calculate the list of changes is not reliable. The team has experienced issues with tags being different accross forks, and such.
+* Installing each package individually is a slow process. There used to be problems with dependencies missing and such that can probably be avoided with the right tooling.
+
+  In the past we had problems because when using workspaces all the dependencies are hoisted so if a dependency was declared in another `package.json`, it will still be found in other packages.
```

Comment:
Suggested: In the past, we have had problems with dependencies declared in one  `package.json` being found in other packages. This happened when using workspaces due to dependencies being hoisted.